app/scripts/metamask-controller.js

```diff
@@ -1507,10 +1506,39 @@ export default class MetamaskController extends EventEmitter {
       signatureController: this.signatureController,
       platform: this.platform,
       extension: this.extension,
-      trackTransactionEvents: handleMMITransactionUpdate.bind(
-        null,
-        transactionMetricsRequest,
-      ),
+      hooks: {
+        getTransactions: this.txController.getTransactions.bind(
+          this.txController,
+        ),
+        updateTransactionHash: this.txController.setTxHash.bind(
+          this.txController,
+        ),
+        trackTransactionEvents: handleMMITransactionUpdate.bind(
+          null,
+          transactionMetricsRequest,
+        ),
+        txStateManager: {
+          getTransactions: this.txController.getTransactions.bind(
+            this.txController,
+          ),
+          setTxStatusSigned:
+            this.txController.txStateManager.setTxStatusSigned.bind(
+              this.txController.txStateManager,
+            ),
+          setTxStatusSubmitted:
+            this.txController.txStateManager.setTxStatusSubmitted.bind(
+              this.txController.txStateManager,
+            ),
+          setTxStatusFailed:
```

Comment:
As discussed, rather than allowing the status to be changed externally, is it safer to introduce a `updateCustodialTransactionStatus` method in core which validates the transaction has a `custodyId` and only allows changing the state to those required, rather than `unapproved` for example.

Reply:
Definitely, done https://github.com/MetaMask/core/pull/2018/commits/ece6d65a61ca6858e1ab9858963a7a9c8b06c553 .